packages/e2e-test-utils/src/widgets.js

```diff
@@ -1,32 +1,29 @@
 /**
  * Internal dependencies
  */
-import { activatePlugin } from './activate-plugin';
-import { deactivatePlugin } from './deactivate-plugin';
-import { visitAdminPage } from './visit-admin-page';
+import { rest, batch } from './rest-api';
 
 /**
  * Delete all the widgets in the widgets screen.
  */
 export async function deleteAllWidgets() {
-	// TODO: Deleting widgets in the new widgets screen is cumbersome and slow.
-	// To workaround this for now, we visit the old widgets screen to delete them.
-	await activatePlugin( 'gutenberg-test-classic-widgets' );
+	const [ widgets, sidebars ] = await Promise.all( [
+		rest( { path: '/wp/v2/widgets' } ),
+		rest( { path: '/wp/v2/sidebars' } ),
+	] );
 
-	await visitAdminPage( 'widgets.php' );
+	await batch(
+		widgets.map( ( widget ) => ( {
+			method: 'DELETE',
+			path: `/wp/v2/widgets/${ widget.id }?force=true`,
+		} ) )
+	);
 
-	let widget = await page.$( '.widgets-sortables .widget' );
-
-	// We have to do this one-by-one since there might be race condition when deleting multiple widgets at once.
-	while ( widget ) {
-		const deleteButton = await widget.$( 'button.widget-control-remove' );
-		const id = await widget.evaluate( ( node ) => node.id );
-		await deleteButton.evaluate( ( node ) => node.click() );
-		// Wait for the widget to be removed from DOM.
-		await page.waitForSelector( `#${ id }`, { hidden: true } );
-
-		widget = await page.$( '.widgets-sortables .widget' );
-	}
-
-	await deactivatePlugin( 'gutenberg-test-classic-widgets' );
+	await batch(
```

Comment:
Could this be batched with the previous batch, or are there concerns about limits? (I think the limit is 25 or something). The limit might actually be a concern here if a test ever adds lots of widgets.

Reply:
We actually can't. I think it might have some race condition going on in those API, I didn't look into why though.

Reply:
Ok 👍 

Might still be worth looking into the batch limit thing (some background - https://github.com/WordPress/gutenberg/issues/28709) as a follow-up, I think this batch function should ideally split the requests array based on the limit when it's over the limit and then make sequential requests for each part.